File: lib_player/src/main/java/ando/player/dialog/BaseDialog.java
package ando.player.dialog;

import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.view.WindowManager;

public class BaseDialog extends Dialog {

    public BaseDialog(Context context, int themeResId) {
        super(context, themeResId);
        initDefaultDialogListener();
    }

    @Override
    public void show() {
        // Set the dialog to not focusable.
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE,
            WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE);

        // Show the dialog with NavBar hidden.
        super.show();

        // Set the dialog to focusable again.
        getWindow().clearFlags(WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE);
    }

    /**
     * 设置dialog消失显示广播
     */
    private void initDefaultDialogListener() {
        setOnShowListener(new OnShowListener() {
            @Override
            public void onShow(DialogInterface dialog) {
            }
        });
        setOnDismissListener(new OnDismissListener() {
            @Override
            public void onDismiss(DialogInterface dialog) {
            }
        });
    }
}

File: lib_player/src/main/java/ando/player/dialog/BaseDialog.kt
package ando.player.dialog

import android.app.Dialog
import android.content.Context
import android.util.Log
import android.view.WindowManager

internal class BaseDialog(context: Context, themeResId: Int) : Dialog(context, themeResId) {

    override fun show() {
        // Set the dialog to not focusable.
        window!!.setFlags(
            WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE,
            WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE
        )
        // Show the dialog with NavBar hidden.
        super.show()
        // Set the dialog to focusable again.
        window!!.clearFlags(WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE)
    }

    /**
     * 设置dialog消失显示广播
     */
    private fun initDefaultDialogListener() {
        setOnShowListener { }
        setOnDismissListener {
            //Log.w("123","Dismiss....................")
        }
    }

    override fun onDetachedFromWindow() {
        super.onDetachedFromWindow()
        //Log.w("123","onDetachedFromWindow....................")
    }

    init {
        initDefaultDialogListener()
    }
}